webapp/components/notifications/NotificationMenu/NotificationMenu.vue

```diff
@@ -35,7 +49,39 @@ export default {
   props: {
     placement: { type: String },
   },
+  created() {
+    setInterval(this.updateNotifications, REFRESH_MILLISEC)
+  },
+  destroyed() {
+    clearInterval(this.updateNotifications)
+  },
   methods: {
+    async updateNotifications() {
+      try {
+        const {
+          data: { notifications },
+        } = await this.$apollo.mutate({
+          mutation: notificationQuery(this.$i18n),
+        })
+        // add all the new notifications to the old notifications at top of the list
+        if (notifications) {
+          notifications.forEach(udatedListNotification => {
+            const sameNotification = this.notifications.find(function(oldListNotification) {
+              return (
+                oldListNotification.from.id === udatedListNotification.from.id &&
+                oldListNotification.createdAt === udatedListNotification.createdAt &&
+                oldListNotification.reason === udatedListNotification.reason
+              )
+            })
+            if (sameNotification === undefined) {
+              this.notifications.unshift(udatedListNotification)
+            }
+          })
+        }
```

Comment:
I find this code way to complex. Can't you just append two arrays and the remove duplicates based on a criteria? I googled and found https://medium.com/dailyjs/how-to-remove-array-duplicates-in-es6-5daa8789641c there is also one by Mozilla using `reduce` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce#Remove_duplicate_items_in_array

Reply:
All your examples @roschaefer are no really working, because if the `read` property changes in my code it should be equal as well. And therefore the new fetched notification should not be added to the beginning …